src/commands/importExportEntities/entities/languages.ts

```diff
@@ -1,11 +1,140 @@
+  'value': { codename }
+})
+
+const createReplaceIsActiveOperation = (isActive: boolean): LanguageModels.IModifyLanguageData => ({
+  'op': 'replace',
+  'property_name': 'is_active',
+  'value': isActive
+})
+
+
+const fixDefaultLanguage = async (client: ManagementClient, projectDefaultLanguage: LanguageModels.LanguageModel, importingDefaultLanguage: LanguageContracts.ILanguageModelContract) => {
+  const operations: LanguageModels.IModifyLanguageData[] = [];
+
+  if (projectDefaultLanguage.codename !== importingDefaultLanguage.codename) {
+    operations.push(createReplaceCodenameOperation(importingDefaultLanguage.codename))
+  }
+
+  if (projectDefaultLanguage.name !== importingDefaultLanguage.name) {
+    operations.push(createReplaceNameOperation(importingDefaultLanguage.name))
+  }
+
+  if (operations.length > 0) {
+    await client
+      .modifyLanguage()
+      .byLanguageId(defaultLanguageId)
+      .withData(operations)
+      .toPromise();
+  }
+}
+
+const importLanguagesToProject = async (client: ManagementClient, importLanguages: ReadonlyArray<LanguageContracts.ILanguageModelContract>) => {
+  const projectLanguages = await client.listLanguages().toAllPromise().then(res => res.data.items);
+
+  for (const lang of importLanguages.filter(l => l.id !== defaultLanguageId)) {
+    const projectLanguage = projectLanguages.find(l => l.codename === lang.codename || l.name === lang.name)
+    if (projectLanguage) {
+      const operations: LanguageModels.IModifyLanguageData[] = [];
+
+      if (!projectLanguage.isActive) {
+        operations.push(createReplaceIsActiveOperation(true));
+      }
+
+      if (projectLanguage.name !== lang.name) {
+        operations.push(createReplaceNameOperation(lang.name))
+      }
+
+      if (projectLanguage.codename !== lang.codename) {
+        operations.push(createReplaceCodenameOperation(lang.codename))
+      }
+
+      if (operations.length > 0) {
+        await client
+          .modifyLanguage()
+          .byLanguageId(lang.id)
+          .withData(operations)
+          .toPromise();
+      }
+
+      continue;
+    }
+
+    const importFallbackLanguage = importLanguages.find(l => l.id === lang.fallback_language?.id);
+
+    if (!importFallbackLanguage) {
+      throw new Error(`Could not find the language to fallback with id ${lang.fallback_language?.id}`)
+    }
+
+    await client
+      .addLanguage()
+      .withData({
+        name: lang.name,
+        codename: lang.codename,
+        is_active: true,
+        external_id: lang.external_id ?? `${lang.codename}-${lang.id}`.slice(0,50),
+      })
+      .toPromise();
+  }
+}
+
+
+const updateFallbackLanguages = async (client: ManagementClient, importLanguages: ReadonlyArray<LanguageContracts.ILanguageModelContract>) => {
+  const projectLanguages = (await client.listLanguages().toAllPromise().then(res => res.data.items));
+
+  for (const lang of importLanguages.filter(l => l.id !== defaultLanguageId)) {
```

Comment:
What would you think about a structure like this instead?
```ts
await serially(importLanguages
  .filter(isNotDefault)
  .map(lang => {
    const projectLanguage = ...;
    if (!projectLanguage) {
      throw new Error(...);
    }
    return [lang, projectLanguage] as const;
  })
  .map(([lang, projectLanguage]) => async () => {
    const projectFallbackLanguage = projectLanguages.find(l => l.id === projectLanguage.fallbackLanguage?.id);
    const importFallbackLanguage = importLanguages.find(l => l.id === lang.fallback_language?.id);
    
    if (...) {
      await ...
    }
  }));
```
It separates the different actions into different method calls (`filter` and `map`s) which should make them a bit more understandable.